### PR TITLE
Add documentation for nested search in knn

### PR DESCRIPTION
**Pull request by heemin32**
### Description
Add documentation for nested search in knn

### Issues Resolved
Fixes https://github.com/opensearch-project/documentation-website/issues/5431

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by heemin32** [comment]
Pending tech review

**Comment by kolchfa-aws** [comment]
Thank you, @heemin32! Let me know when you're ready for me do to a doc review on this PR.

File: _search-plugins/knn/nested-search-knn.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+nav_order: 21
+parent: k-NN search
+grand_parent: Search methods
+has_children: false
+has_math: true
+---
+
+# k-NN search with nested fields
+With k-NN search using nested fields, you can store multiple vectors in a single document and search for the k-nearest documents. 
+Each document is scored using the nearest vector in the nested field.
```
Comment (martin-gaievski) [inline _search-plugins/knn/nested-search-knn.md:13]: this phase is a bit misleading. It doesn't say if the whole document is scored based on a best score from all the vectors from its nested field(s). Need more details on what is `nearest vector` in the context of this sentence.

Comment (heemin32) [reply _search-plugins/knn/nested-search-knn.md:13]: Right. It could be or could not be the best score from all the vectors from its nested field. Let me rephrase it.

File: _search-plugins/knn/nested-search-knn.md [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+parent: k-NN search
+grand_parent: Search methods
+has_children: false
+has_math: true
+---
+
+# k-NN search with nested fields
+With k-NN search using nested fields, you have the flexibility to store multiple vectors within a single document.
+For instance, if your document comprises various components, you can generate a vector value for each component and store them in a nested field within the document.
+When performing a search with a k value of 2, even if a single document has 2 nearest vectors, the search results will include 2 documents: one with the nearest vector and another with the third nearest vector.
```
Comment (martin-gaievski) [inline _search-plugins/knn/nested-search-knn.md:14]: Can this sentence be rephrased into a more simpler form? Maybe first state the overall high-level idea that the document will be included to the result only once and that is based on the highest score/most similarity of all vectors stored in all its nested fields. Example will follow, something like: index has 2 docs, doc A has two fields with vector A1 and A2, and doc B with vector B1. When we run knn search with k == 2. Most similar vectors are A1, A2, B1, but final result will be [A, B] because each doc id will be unique.

**Review by martin-gaievski — APPROVED** [review]
Looks good to me

**Review by jmazanec15 — APPROVED** [review]
LGTM

**Comment by heemin32** [comment]
@kolchfa-aws Could you review this PR?

**Comment by kolchfa-aws** [comment]
Sure, @heemin32. I'll review by the end of the week.

**Comment by kolchfa-aws** [comment]
@heemin32 Please review my [commit](https://github.com/opensearch-project/documentation-website/pull/6237/commits/451083aef38e3e19c36403556bb58009f9f0264d) to ensure that it preserves technical accuracy. Then we'll be ready to move this PR to editorial review. Thanks!

**Review by heemin32 — COMMENTED** [review]
LGTM

File: _search-plugins/knn/approximate-knn.md [hunk 3]
```diff
@@ -246,6 +246,10 @@ After data is ingested, it can be search just like any other `knn_vector` field!
 
 To learn about using filters with k-NN search, see [k-NN search with filters]({{site.url}}{{site.baseurl}}/search-plugins/knn/filter-search-knn/).
 
+### Using approximate k-NN with nested fields
+
+To learn about k-NN search with nested fields, see [k-NN search with nested fields]({{site.url}}{{site.baseurl}}/search-plugins/knn/nested-search-knn/).
```
Comment (natebower) [inline _search-plugins/knn/approximate-knn.md:251]:
```suggestion
To learn about using k-NN search with nested fields, see [k-NN search with nested fields]({{site.url}}{{site.baseurl}}/search-plugins/knn/nested-search-knn/).
```

File: _search-plugins/knn/nested-search-knn.md [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+grand_parent: Search methods
+has_children: false
+has_math: true
+---
+
+# k-NN search with nested fields
+
+Using [nested fields]({{site.url}}{{site.baseurl}}/field-types/nested/) in a k-NN index, you can store multiple vectors in a single document. For example, if your document consists of various components, you can generate a vector value for each component and store each vector in a nested field.
+
+A k-nearest neighbors document search operates at the field level. For a document with nested fields, OpenSearch examines only the vector nearest to the query vector to decide whether to include the document in the results. For example, consider an index containing documents `A` and `B`. Document `A` is represented by vectors `A1` and `A2`, and document `B` is represented by vector `B1`. Further, the similarity order for a query Q is `A1`, `A2`, `B1`. If you search using query Q with a k value of 2, the search will return both documents `A` and `B` instead of only document `A`.
```
Comment (natebower) [inline _search-plugins/knn/nested-search-knn.md:15]:
```suggestion
A k-NN document search operates at the field level. For a document with nested fields, OpenSearch examines only the vector nearest to the query vector to decide whether to include the document in the results. For example, consider an index containing documents `A` and `B`. Document `A` is represented by vectors `A1` and `A2`, and document `B` is represented by vector `B1`. Further, the similarity order for a query Q is `A1`, `A2`, `B1`. If you search using query Q with a k value of 2, the search will return both documents `A` and `B` instead of only document `A`.
```

File: _search-plugins/knn/nested-search-knn.md [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+
+Note that in the case of an approximate search, the results are approximations and not exact matches.
+
+k-NN search with nested fields is supported by the HNSW algorithm for the Lucene and the Faiss engines.
```
Comment (natebower) [inline _search-plugins/knn/nested-search-knn.md:19]:
```suggestion
k-NN search with nested fields is supported by the HNSW algorithm for the Lucene and Faiss engines. 
```

File: _search-plugins/knn/nested-search-knn.md [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+```json
+PUT _bulk?refresh=true
+{ "index": { "_index": "my-knn-index-1", "_id": "1" } }
+{"nested_field":[{"my_vector":[1,1,1]},{"my_vector":[2,2,2]},{"my_vector":[3,3,3]}]}
+{ "index": { "_index": "my-knn-index-1", "_id": "2" } }
+{"nested_field":[{"my_vector":[10,10,10]},{"my_vector":[20,20,20]},{"my_vector":[30,30,30]}]}
+```
+{% include copy-curl.html %}
+
+Then run a k-nearest neighbor search on the data using the `knn` query type:
```
Comment (natebower) [inline _search-plugins/knn/nested-search-knn.md:71]:
```suggestion
Then run a k-NN search on the data by using the `knn` query type:
```

File: _search-plugins/knn/nested-search-knn.md [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+          }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Even though all three nearest vectors to the query vector are in document 1, the query returns both documents 1 and 2 because k is set to 2:
```
Comment (natebower) [inline _search-plugins/knn/nested-search-knn.md:93]:
```suggestion
Even though all three vectors nearest to the query vector are in document 1, the query returns both documents 1 and 2 because k is set to 2:
```

File: _search-plugins/knn/nested-search-knn.md [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
+        }
+      }
+    ]
+  }
+}
+```
+
+## k-NN search with filtering on nested fields
+
+You can apply a filter to k-NN search with nested fields. A filter can be applied to either a top-level field or a field inside a nested field.
```
Comment (natebower) [inline _search-plugins/knn/nested-search-knn.md:179]:
```suggestion
You can apply a filter to a k-NN search with nested fields. A filter can be applied to either a top-level field or a field inside a nested field.
```

File: _search-plugins/knn/nested-search-knn.md [hunk 9]
```diff
@@ -0,0 +1,347 @@
...
+{ "index": { "_index": "my-knn-index-1", "_id": "1" } }
+{"parking": false, "nested_field":[{"my_vector":[1,1,1]},{"my_vector":[2,2,2]},{"my_vector":[3,3,3]}]}
+{ "index": { "_index": "my-knn-index-1", "_id": "2" } }
+{"parking": true, "nested_field":[{"my_vector":[10,10,10]},{"my_vector":[20,20,20]},{"my_vector":[30,30,30]}]}
+{ "index": { "_index": "my-knn-index-1", "_id": "3" } }
+{"parking": true, "nested_field":[{"my_vector":[100,100,100]},{"my_vector":[200,200,200]},{"my_vector":[300,300,300]}]}
+```
+{% include copy-curl.html %}
+
+Then run a k-nearest neighbors search on the data using the `knn` query type with a filter. The following query returns documents whose `parking` field is set to `true`:
```
Comment (natebower) [inline _search-plugins/knn/nested-search-knn.md:232]:
```suggestion
Then run a k-NN search on the data using the `knn` query type with a filter. The following query returns documents whose `parking` field is set to `true`:
```

File: _search-plugins/knn/nested-search-knn.md [hunk 10]
```diff
@@ -0,0 +1,347 @@
...
+          }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Even though all three nearest vectors to the query vector are in document 1, the query returns documents 2 and 3 because document 1 is filtered out:
```
Comment (natebower) [inline _search-plugins/knn/nested-search-knn.md:263]:
```suggestion
Even though all three vectors nearest to the query vector are in document 1, the query returns documents 2 and 3 because document 1 is filtered out:
```

File: _search-plugins/knn/nested-search-knn.md [hunk 11]
```diff
@@ -0,0 +1,347 @@
...
+nav_order: 21
+parent: k-NN search
+grand_parent: Search methods
+has_children: false
+has_math: true
+---
+
+# k-NN search with nested fields
+
+Using [nested fields]({{site.url}}{{site.baseurl}}/field-types/nested/) in a k-NN index, you can store multiple vectors in a single document. For example, if your document consists of various components, you can generate a vector value for each component and store each vector in a nested field.
```
Comment (natebower) [inline _search-plugins/knn/nested-search-knn.md:13]: Should we define k-NN on first appearance in this para?

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _search-plugins/knn/nested-search-knn.md (hunk 11 above)
Comment (kolchfa-aws) [inline _search-plugins/knn/nested-search-knn.md:13]:
```suggestion
Using [nested fields]({{site.url}}{{site.baseurl}}/field-types/nested/) in a k-nearest neighbors (k-NN) index, you can store multiple vectors in a single document. For example, if your document consists of various components, you can generate a vector value for each component and store each vector in a nested field.
```

**Review by kolchfa-aws — APPROVED** [review]

**Comment by dylan-tong-aws** [comment]
Users are going to be looking for info on how to perform vector search on long documents or "vector chunks". They will not be looking for how to "use  k-NN search with nested fields". We need to connect the dots for the users/
